proposals/0759-react-native-frameworks.md

```diff
@@ -0,0 +1,452 @@
+
+#### DON’Ts of Frameworks
+
+- **Do not hard fork react-native** - As a rule of thumb, you should not GitHub-fork react-native and re-publish it. If you’re doing so, it’s probably due to a lack of APIs that React Native should provide. Please reach out to us and attempt to de-fork your framework.
+- **Do not soft fork the core APIs** - We want to make sure developers need to learn only once how to write a native component. Building abstraction and wrappers is accepted, forking is not, as it makes it harder for users to share components/modules between targets.
+- **Do not fork the ecosystem** - It’s crucial that `react-native-*` libraries can be easily shared and ported to other targets or frameworks (with little to no work). Frameworks should aim at building libraries that work across the whole ecosystem and not solutions that silo the users away.
+- **Do not redistribute prebuilts** - React Native ships native prebuilts for several targets. Do not attempt to republish them as this reduces security for users and is essentially yet another fork.
+- **Do not depend on the internals** - Do not depend on internal APIs. As a rule of thumb, depending on private/internal API (i.e. those from the `internal` package or with `unstable_` prefix) is not recommended and will cause breakages in the future. Specifically, we might consider rotating private APIs (changing the name) in the future to discourage its usages. When there is a need for Frameworks to access those internal APIs, we’ll be annotating them to make them easy to recognize it (i.e. with the `@FrameworkAPI` annotation).
+
+#### Note for Platforms developers
+
+The aforementioned Do/Dont’s don’t apply strictly to **Platforms developers**. While they do represent general good practices in the software development field, targeting a new platform poses a series of **unique challenges**.
+
+Specifically, when supporting platforms, having a fork might actually be the preferred solution, based on the amount of custom logic that is necessary to maintain. Similarly, certain platforms require redistributing prebuilts for new binary architectures which are not already provided by React Native.
+
+The React Native team is committed to fostering innovation and supporting new platforms by offering APIs and artifacts for those new platforms, whenever possible.
+
+## Evolving the React Native Community Template & CLI
+
+Users who have been using the React Native **Community CLI & Template**, might be wondering how React Native Frameworks affect them.
+
+We want to reassure our users that are using those tools and clarify that those tools will **still be supported** **and updated**, and you can keep on using them if you wish. However, we’ll be recommending new users and **newly created mobile apps** to use a React Native Framework from now on. Similarly, you’re free to migrate to a React Native Framework if you intend to.
```

Comment:
super important point here for many folks who will come across the rfc! this rfc has an absolutely minimal impact on developers that want to use `@react-native-community/cli` tooling - effectively, when the changes roll out:

- initialize new apps with `npx @react-native-community/cli init` instead of `npx react-native init` if you prefer community tooling
- `npx react-native config` becomes `npx @react-native-community/cli config` and `npx react-native doctor` becomes something like `npx react-native-doctor`. commands like `run-ios` and `start` are considered to be part of `react-native` itself, as discussed below